test/integration/consul-container/test/README.md

```diff
@@ -0,0 +1,123 @@
+# Upgrade Integration Tests
+
+The upgrade integration test is responsible for ensuring that users are able to upgrade between consul versions without experiencing difficulties. The upgrade tests does the following:
+- Ensure that customers are able to upgrade to a new Consul version, and new and existing features work as expected. 
+- This tests aims to ensure a high degree of interoperability between versions of Consul by supporting `n-2` and `n-1` upgrades, ensuring that upgrading to newer versions does not degrade or break functionalities.
+    - Validate users are able to perform a direct upgrade of Consul. For example upgrading from `1.13 → 1.14`.  
+    - Validate users are able to perform skip-version upgrade of Consul. For example upgrading from `1.13 → 1.15`.   
+
+- The upgrade tests also aims to highlight errors that may occur as users attempt to upgrade their current version to a newer version. 
+
+Here is an example of how the upgrade tests work
+1. Create a cluster with a specified number of server and client agents, then enable the feature to be tested. 
+2. Create some workload in the cluster, e.g. register 2 services: static-server, static-client. 
+3. Verify connection / disconnection (e.g. deny Action) 
+4. Upgrade consul cluster and restart the envoy sidecars (we restart envoy sidecar to ensure the upgraded consul binary can read the state from the previous version and generate the correct envoy configurations) 
+5. Verify connection / disconnection (e.g., deny Action)  
+
+**Note** that all consul agents and user workloads such as application services, mesh-gateway are running in docker containers.
+
+
+The tests focus on Consul upgrade lifecycle and emit logs to stdout which developers can silence. Please reach out to the CTIA Team for any questions or help getting started.
```

Comment:
I'm kind of struggling to relate these clauses?

> These tests focus on Consul upgrade lifecycle.

Probably redundant with the intro paragraph?

> [Containers] emit logs to stdout which developers can silence [with `--follow-logs=false`].

Kind of a detail for later IMO

> Please reach out to the CTIA Team for any questions or help getting started.

Is it ok to put internal team names in public-facing docs? I honestly am not sure.